app/views/posts/index.html.erb

```diff
@@ -1 +1,18 @@
-<p> list of all posts and comment of a user </p>
\ No newline at end of file
+<h2> list of all posts and comment of a user </h2>
+
+ <div class = 'posts-content'>
+     <% @posts.each do |post|%>
+     <div class = 'user-post'>
+    <%= render partial: 'post', locals: { post: post } %>
+     <div class='comments-content'>
+     <p><strong> Comments :</strong> </p>
+     <% post.recent_comments.each do |comment|%>
```

Comment:
- [Optional] Hi @AmineDerbal, I'm so impressed because you almost implemented the requested changes plus optional changes. You utilize the `#recent_comments` method as expected :+1: but there's a hectic here: This method is not working as expected. This method should render the 5 most recent comments. Please NOT 3. kindly update this method in the Post model to `comments.order(created_at: :desc).limit(5)` Please just change 3 to 5 before merging.

Reply:
Understood